t/control/plugin-metadata.t

```diff
@@ -0,0 +1,124 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+use t::APISIX 'no_plan';
+
+repeat_each(1);
+no_long_string();
+no_root_location();
+no_shuffle();
+log_level("info");
+
+run_tests;
+
+__DATA__
+
+=== TEST 1: add plugin metadatas
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local _, body = t('/apisix/admin/plugin_metadata/example-plugin',
+                ngx.HTTP_PUT,
+                [[{
+                    "skey": "val",
+                    "ikey": 1
+                }]],
+                [[{
+                    "node": {
+                        "value": {
+                            "skey": "val",
+                            "ikey": 1
+                        }
+                    },
+                    "action": "set"
+                }]]
+                )
+
+            ngx.say(body)
+
+            local _, body = t('/apisix/admin/plugin_metadata/file-logger',
+                ngx.HTTP_PUT,
+                [[
+                {"log_format": {"upstream_response_time": "$upstream_response_time"}}
+                ]]
+                )
+            ngx.say(body)
+        }
+    }
+--- request
```

Comment:
We can set the repeated sections in `add_block_preprocessor` like the other test files:
https://github.com/apache/apisix/blob/44d20c391767d3d42c5111ea7ef6318e84633996/t/deployment/conf_server.t#L28

Reply:
@kingluo Please update the test